### PR TITLE
Fixing several code issues

### PR DESCRIPTION
**Pull request by clanghout**
Fixing some deprecated calls.
Fixing some stuff SonarLint reported to me.
Update dependencies to latest.

create gridWrapper in an attempt to split some muuri implementation from the gridcontrols file.

### CODE REVIEW

File: src/gridControls.ts [hunk 1]
```diff
@@ -349,20 +281,20 @@ export class CTabGrid {
             }
         });
 
-    };
+    }
 
-    // Retrieve the current config from the browser's localstorage
-    public getConfig(): CTabWidgetSerialized[] {
+    // Retrieve the current config from the browser"s localstorage
```
Comment (foresterre) [inline src/gridControls.ts:286]: " -> '

Comment (foresterre) [inline src/gridControls.ts:286]: " -> '

File: src/gridControls.ts [hunk 2]
```diff
@@ -411,16 +343,16 @@ export class CTabGrid {
             console.log(widgetTypes);
             console.error(e);
         }
-    };
+    }
 
     // When dragging stops, we update the ordering indices for all elements
     // FIXME: This does mean that if we are in a different ordering view (e.g.
-    //   "date added", we'll update the ordering to the currently active view);
+    //   "date added", we"ll update the ordering to the currently active view);
```
Comment (foresterre) [inline src/gridControls.ts:350]: " -> '

File: src/gridControls.ts [hunk 3]
```diff
@@ -466,12 +398,12 @@ export class CTabGrid {
...
                 matchingWidget.settings.orderIndex = index;
                 // this is the value used by the grid for sorting
-                widgetBody.dataset.orderIndex = index;
+                (widgetBody as HTMLElement).dataset.orderIndex = `${index}`;
 
                 // we need to reset the sorting data, since we updated the orderIndex, used by one of the sorters
                 updatedSortingData = true;
             } else {
-                console.warn(`Didn't find a matching widget for ${widgetBody.id}`);
+                console.warn(`Didn"t find a matching widget for ${widgetBody!.id}`);
```
Comment (foresterre) [inline src/gridControls.ts:406]: " -> '

**Comment by clanghout** [comment]
- [ ] fix the remove option

**Comment by foresterre** [comment]
Fixed the remove not working in 64f507b

**Pull request merged**